src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -129,7 +130,8 @@ module Make (Inputs : Inputs_intf) :
         kill processed_transition_reader processed_transition_writer ;
         kill catchup_job_reader catchup_job_writer ;
         kill catchup_breadcrumbs_reader catchup_breadcrumbs_writer ;
-        kill proposer_transition_reader_copy proposer_transition_writer_copy )
+        kill proposer_transition_reader_copy proposer_transition_writer_copy ;
+        Ivar.fill_if_empty clean_up_catchup_scheduler () )
```

Comment:
Can't this just be `Ivar.fill`? Pretty certain it can be like this.

Reply:
It should be. I would change it.